odo/backends/sql.py

```diff
@@ -172,7 +172,9 @@ def discover_typeengine(typ):
     if isinstance(typ, (sa.NUMERIC, sa.DECIMAL)):
         return datashape.Decimal(precision=typ.precision, scale=typ.scale)
     if isinstance(typ, (sa.String, sa.Unicode)):
-        return datashape.String(typ.length, typ.collation)
+        return datashape.String(typ.length, None)
+    if isinstance(typ, sa.dialects.mssql.BIT):
+        return datashape.int_
```

Comment:
for things like bit here which do not need to do any conditional construction I think it would be okay to just add this to the `revtypes.update(...)` mapping. To simplify the code path here if we have a lot of types that need some specialization we could use mulipledispatch to do something like:

```python
@dispatch(type)
def from_sql_type(type_):
    if type_ in revtypes:
        return dshape(revtypes[type_])[0]
    # body of the else in the current function
    # for k, v in revtypes.items():
    #     ...

@dispatch(object)
def from_sql_type(ob):
    return _from_sql_type(type(ob))

@dispatch((sa.String, sa.Unicode))
def from_sql_type(type_):
    return datashape.String(type_.length, None)

...
```

This might help us add more special handling in the future without requiring us to make this function a nasty if/else mess. Depending on what we decide we could even document this function to allow users to register their own custom sqlalchemy type converters!

Does this approach make sense or do you think it is overkill for something like this?